### PR TITLE
Change ProductionServerURI to https://api.keybase.io

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco

### CODE REVIEW

**Comment by maxtaco** [comment]
what steps did you take to test?

**Comment by patrickxb** [comment]
Nevermind.  api.keybase.io doesn't work currently.

**Comment by patrickxb** [comment]
I thought it was already set up based on what we discussed yesterday.

**Comment by maxtaco** [comment]
WE should do a bunch of testing here:
  * that everything still works when hitting the live servers, since now we're using our CA bundle and previously we didn't
  * that everything breaks when we intentionally break our CA bundle.  maybe substitute in a test dummy CA bundle that was generated for testing

**Comment by patrickxb** [comment]
@maxtaco I added tests for ping with bundled CA and with an unknown CA.

File: go/libkb/api_test.go [hunk 1]
```diff
@@ -24,3 +26,143 @@ func TestIsReddit(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	_, err = tc.G.API.Get(arg)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestProductionBadCA(t *testing.T) {
```
Comment (maxtaco) [inline go/libkb/api_test.go]: great test

File: go/libkb/api_test.go [hunk 2]
```diff
@@ -24,3 +26,143 @@ func TestIsReddit(t *testing.T) {
...
+
+	t.Log("WARNING: setting run mode to production, be careful:")
+	tc.G.Env.Test.UseProductionRunMode = true
+
+	if tc.G.Env.GetServerURI() != uriExpected {
+		t.Fatalf("production server uri: %s, expected %s", tc.G.Env.GetServerURI(), uriExpected)
+	}
+
+	// change the api CA to one that api.keybase.io doesn't know:
+	BundledCAs["api.keybase.io"] = unknownCA
```
Comment (maxtaco) [inline go/libkb/api_test.go]: this is slightly worrisome since you never change it back

Comment (maxtaco) [inline go/libkb/api_test.go]: probably the cleanest way to do this is to make `Env` fetch this as a test parameter, no? As we do some of the other parameters. That way you don't have to worry about crashing before you set it back.  Alternatively, you can just set it back with `defer`

Comment (patrickxb) [inline go/libkb/api_test.go]: fixed

Comment (patrickxb) [inline go/libkb/api_test.go]: I used defer before reading your comment.  I can change to Env pattern if you prefer.

Comment (maxtaco) [inline go/libkb/api_test.go]: i'd add in `Env.Test`.  Seems like you use it just above anyways.

Comment (patrickxb) [inline go/libkb/api_test.go]: Ok.  This does test that we're actually using that map, however, so we'll lose that coverage.

Comment (maxtaco) [inline go/libkb/api_test.go]: Yeah that's the other side of it.... Ok, then leave it as is....

**Comment by maxtaco** [comment]
:+1:

**Comment by maxtaco** [comment]
(after CI fixes)

**Comment by patrickxb** [comment]
I can't get github to add commit 061ebcc47b0c71219de0998729ef638ae0bafebe to this PR.  It shows up in the branch:  https://github.com/keybase/client/commits/patrickxb/CORE-2857 however.

I know github was down, so maybe that has something to do with it.

Tried rebase of master plus force push...

**Comment by maxtaco** [comment]
Github should be back up now

**Comment by maxtaco** [comment]
https://status.github.com/

**Comment by patrickxb** [comment]
I know, but it still isn't including that commit in this PR.

**Comment by maxtaco** [comment]
I see that commit above now...

**Comment by patrickxb** [comment]
yes, it looks like it is showing up now...

just another round of CI waiting.

**Pull request merged**